src/pdb/mod.rs

```diff
@@ -207,6 +207,9 @@ impl PageFlags {
 #[br(little, magic = 0u32)]
 #[br(import(page_size: u32))]
 pub struct Page {
+    /// Stream position immediately after magic; used to compute heap base for standalone buffers.
+    #[br(temp, parse_with = current_offset)]
+    page_pos_after_magic: u64,
```

Comment:
consider reordering here so you first record the temp page_pos and then assert the magic in a separate private field. That would avoid having to mess with questionable index shifts like `- 4`